### PR TITLE
feat: this bumps sqlite up to 3.40.1 [fixes VM-223]

### LINKED ISSUE #6: Fix issue with FTS5
After the BWU refactoring, SQLite starts crashing on a virtual table creating with `FTS5`.

The reproducible example:
```sql
CREATE VIRTUAL TABLE users USING FTS5(body)
```

### PR DESCRIPTION
**Pull request by raftedproc**
Fixes #6

### CODE REVIEW

File: src/vdbeapi.c [hunk 1]
```diff
@@ -1108,53 +1197,52 @@ static void columnMallocFailure(sqlite3_stmt *pStmt)
...
-sqlite_int64 sqlite3_column_int64(sqlite3_stmt *pStmt, int i){
-  sqlite_int64 val = sqlite3_value_int64( columnMem(pStmt,i) );
+sqlite_int64 sqlite3_column_int64(sqlite3_stmt *pStmt, int i) {
+  sqlite_int64 val = sqlite3_value_int64(columnMem(pStmt, i));
   columnMallocFailure(pStmt);
   return val;
 }
 
-void sqlite3_column_text_(sqlite3_stmt *pStmt, int i) __EXPORT_NAME(sqlite3_column_text) {
+void sqlite3_column_text_(sqlite3_stmt *pStmt, int i)
```
Comment (mikevoronov) [inline src/vdbeapi.c:1244]: let's wrap this function with `#ifndef __sqlite_unmodified_upstream`

File: src/vdbeapi.c [hunk 2]
```diff
@@ -1108,53 +1197,52 @@ static void columnMallocFailure(sqlite3_stmt *pStmt)
 ** The following routines are used to access elements of the current row
 ** in the result set.
 */
-
-void sqlite3_column_blob_(sqlite3_stmt *pStmt, int i) __EXPORT_NAME(sqlite3_column_blob) {
+void sqlite3_column_blob_(sqlite3_stmt *pStmt, int i) __attribute__((export_name("sqlite3_column_blob"))) {
```
Comment (mikevoronov) [inline src/vdbeapi.c:1200]: let's wrap this function with `#ifndef __sqlite_unmodified_upstream`

File: src/main.c [hunk 3]
```diff
@@ -2449,7 +2574,7 @@ int sqlite3TempInMemory(const sqlite3 *db){
 ** Return UTF-8 encoded English language explanation of the most recent
 ** error.
 */
-void sqlite3_errmsg_(sqlite3 *db) __EXPORT_NAME(sqlite3_errmsg) {
+void sqlite3_errmsg_(sqlite3 *db) __attribute__((export_name("sqlite3_errmsg"))) {
```
Comment (mikevoronov) [inline src/main.c:2577]: let's wrap this function with `#ifndef __sqlite_unmodified_upstream`

**Review by mikevoronov — APPROVED** [review]
utACK.
Good job!

File: src/vdbeapi.c (hunk 1 above)
Comment (raftedproc) [reply src/vdbeapi.c:1244]: Nice catch.

File: src/vdbeapi.c (hunk 2 above)
Comment (raftedproc) [reply src/vdbeapi.c:1200]: Nice catch.

**Pull request merged**